Python/flowgraph.c

```diff
@@ -1602,6 +1602,7 @@ optimize_basic_block(PyObject *const_cache, basicblock *bb, PyObject *consts)
                 }
                 break;
             case JUMP:
+            case JUMP_NO_INTERRUPT:
```

Comment:
What about the opposite case, `JUMP` jumping to `JUMP_NO_INTERRUPT?
Now that `JUMP_NO_INTERRUPT` is more common, we should handle that case as well.

Reply:
I considered this, but the new JUMP_NO_INTERRUPT have no line number, so the current threading logic wouldn't do anything (unless the following jump also has no line number, which is unlikely). So the jump_thread function needs to change as well. I can do it in the same PR, if you don't think it complicates the review.

Reply:
It should be a fairly straightforward simple change to `jump_thread`.
If the initial jump has no line number, use the line number of the target.

Might as well do it this PR.

Reply:
Once I add that I get assertion failed about some redundant jump not being cleared up, so I think that exposes another problem.

Reply:
The problem was that the assertion in no_redundant_jumps wasn't checking line numbers, so it tripped when it shouldn't have. All good now.